File: examples/cache.py
"""Application cache demo (sync + async, default driver: moka).

    python examples/cache.py
"""

from __future__ import annotations

import asyncio

from fusion_framework import cache


async def async_demo() -> None:
    await cache.aset("async-greeting", {"hello": "async"}, ttl=30)
    print("aget:", await cache.aget("async-greeting"))
    print(
        "aget_or_set:",
        await cache.aget_or_set("async-counter", _async_one),
    )
    await cache.aclear()
    print("after aclear:", await cache.aget("async-greeting"))


async def _async_one() -> int:
    await asyncio.sleep(0)
    return 1


def main() -> None:
    cache.configure_driver("moka", default_ttl=60)
    cache.set("greeting", {"hello": "world"}, ttl=30)
    print("get:", cache.get("greeting"))
    print("exists:", cache.exists("greeting"))
    print("get_or_set:", cache.get_or_set("counter", lambda: 1))
    print("exists_or_set (first):", cache.exists_or_set("flag", True))
    print("exists_or_set (again):", cache.exists_or_set("flag", False))
    print("delete_or_set:", cache.delete_or_set("greeting", {"hello": "fusion"}))
    print("driver:", cache.driver())
    cache.clear()
    print("after clear:", cache.get("greeting"))
    asyncio.run(async_demo())


if __name__ == "__main__":
    main()

File: examples/cache.mjs
/**
 * Application cache demo (sync + async, default driver: moka).
 *
 *   node examples/cache.mjs
 */
import { cache } from 'fusion-framework'

cache.configureDriver('moka', { defaultTtl: 60 })
cache.set('greeting', { hello: 'world' }, 30)
console.log('get:', cache.get('greeting'))
console.log('exists:', cache.exists('greeting'))
console.log('getOrSet:', cache.getOrSet('counter', () => 1))
console.log('existsOrSet (first):', cache.existsOrSet('flag', true))
console.log('existsOrSet (again):', cache.existsOrSet('flag', false))
console.log('deleteOrSet:', cache.deleteOrSet('greeting', { hello: 'fusion' }))
console.log('driver:', cache.driver())
cache.clear()
console.log('after clear:', cache.get('greeting'))

await cache.aset('async-greeting', { hello: 'async' }, 30)
console.log('aget:', await cache.aget('async-greeting'))
console.log(
  'agetOrSet:',
  await cache.agetOrSet('async-counter', async () => 1),
)
await cache.aclear()
console.log('after aclear:', await cache.aget('async-greeting'))
